contracts/libraries/VaultLifecycleSTETH.sol

```diff
@@ -0,0 +1,794 @@
+            otoken.collateralAsset() == collateralAsset,
+            "Wrong collateralAsset"
+        );
+
+        // we just assume all options use USDC as the strike
+        require(otoken.strikeAsset() == USDC, "strikeAsset != USDC");
+
+        uint256 readyAt = block.timestamp.add(delay);
+        require(otoken.expiryTimestamp() >= readyAt, "Expiry before delay");
+    }
+
+    function rollover(
+        uint256 currentSupply,
+        uint256 currentBalance,
+        Vault.VaultParams calldata vaultParams,
+        Vault.VaultState calldata vaultState
+    )
+        external
+        pure
+        returns (
+            uint256 newLockedAmount,
+            uint256 queuedWithdrawAmount,
+            uint256 newPricePerShare,
+            uint256 mintShares
+        )
+    {
+        uint256 pendingAmount = uint256(vaultState.totalPending);
+        uint256 roundStartBalance = currentBalance.sub(pendingAmount);
+
+        uint256 singleShare = 10**uint256(vaultParams.decimals);
+
+        newPricePerShare = getPPS(
+            currentSupply,
+            roundStartBalance,
+            singleShare
+        );
+
+        // After closing the short, if the options expire in-the-money
+        // vault pricePerShare would go down because vault's asset balance decreased.
+        // This ensures that the newly-minted shares do not take on the loss.
+        uint256 _mintShares =
+            pendingAmount.mul(singleShare).div(newPricePerShare);
+
+        uint256 newSupply = currentSupply.add(_mintShares);
+        // TODO: We need to use the pps of the round they scheduled the withdrawal
+        // not the pps of the new round. https://github.com/ribbon-finance/ribbon-v2/pull/10#discussion_r652174863
+        uint256 queuedWithdrawAmount =
+            newSupply > 0
+                ? uint256(vaultState.queuedWithdrawShares)
+                    .mul(currentBalance)
+                    .div(newSupply)
+                : 0;
+
+        uint256 balanceSansQueued = currentBalance.sub(queuedWithdrawAmount);
+
+        return (
+            balanceSansQueued,
+            queuedWithdrawAmount,
+            newPricePerShare,
+            _mintShares
+        );
+    }
+
+    // https://github.com/opynfinance/GammaProtocol/blob/master/contracts/Otoken.sol#L70
+    uint256 private constant OTOKEN_DECIMALS = 10**8;
+
+    function createShort(
+        address gammaController,
+        address marginPool,
+        address oTokenAddress,
+        uint256 depositAmount
+    ) external returns (uint256) {
+        IController controller = IController(gammaController);
+        uint256 newVaultID =
+            (controller.getAccountVaultCounter(address(this))).add(1);
+
+        IOtoken oToken = IOtoken(oTokenAddress);
+        address collateralAsset = oToken.collateralAsset();
+
+        uint256 collateralDecimals =
+            uint256(IERC20Detailed(collateralAsset).decimals());
+        uint256 mintAmount;
+
+        if (oToken.isPut()) {
```

Comment:
Does it make sense to remove all conditions checking `isPut`? stETH can only be used for call options.

Cons of doing so: The code across VaultLifecycles are not consistent

Reply:
we can remove it